### DISCUSSION TITLE
Distributed upgrade to 2.4?

### DISCUSSION THREAD

**Discussion by rosswakelin** [Security-Onion-Solutions/securityonion#10044] [2.4]
I know its like asking how long is a piece of string, but is there an intended timeframe for the 2.4 production release?  We are just about to install and ship 25 sensor nodes to locations thousands of kilometres away, and the statement that there is no in-place upgrade from 2.4 to 2.3 is scary. We might be able to delay the build/shipping for a short while but we need some idea of how long.  
The other question is around migrating from a currently running 2.3 grid to 2.4, keeping things like existing data, alerts and config files.  Any plans for this yet?

Thanks

Ross

**Comment by InfosecGoon** [comment]
As posted in the blog post announcing the 2.4 beta, we're still working through the possibilities for an upgrade path. That said, if you're building these sensor nodes in-house, I would recommend making sure that /nsm is on its own partition so there's separation between the operating system and the Security Onion data.

**Reply by rosswakelin** [reply]
Thanks for that
The standard SO installer does not give you the option of creating your own partition table.  If you have two disks is gives you the option of putting /nsm on a second disk, but if you only have one disk it just goes ahead and does its own thing.

**Reply by InfosecGoon** [reply]
It puts them in separate LVM pools though, right? System-root vs. system-nsm?
